Flaky-test: NonPersistentTopicTest.testProducerRateLimit

### Search before reporting

- [x] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.


### Example failure

https://github.com/apache/pulsar/actions/runs/18779396830/job/53582346812#step:11:1392

### Exception stacktrace

```text
[ERROR] org.apache.pulsar.client.api.NonPersistentTopicTest.testProducerRateLimit -- Time elapsed: 2.613 s <<< FAILURE!
java.lang.AssertionError:

Expecting any element of:
  [ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="test", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.managed_ledger.message.outgoing.replicated.size, description=The total number of messages bytes written to this ledger, including replicas., unit=By, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1761309361708886628, epochNanos=1761309362952485121, attributes={pulsar.managed_ledger.name="my-property/my-ns/persistent/__change_events", pulsar.namespace="my-property/my-ns"}, value=0, exemplars=[]}], monotonic=true, aggregationTemporality=CUMULATIVE}},
    ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="test", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.managed_ledger.cache.entry.count, description=The number of entries in the entry cache., unit={entry}, type=LONG_SUM, data=ImmutableSumData{points=[ImmutableLongPointData{startEpochNanos=1761309361708886628, epochNanos=1761309362952485121, attributes={pulsar.managed_ledger.cache.entry.status="active"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1761309361708886628, epochNanos=1761309362952485121, attributes={pulsar.managed_ledger.cache.entry.status="inserted"}, value=0, exemplars=[]}, ImmutableLongPointData{startEpochNanos=1761309361708886628, epochNanos=1761309362952485121, attributes={pulsar.managed_ledger.cache.entry.status="evicted"}, value=0, exemplars=[]}], monotonic=false, aggregationTemporality=CUMULATIVE}},
    ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={pulsar.cluster="test", service.name="pulsar-broker", service.version="4.2.0-SNAPSHOT", telemetry.sdk.language="java", telemetry.sdk.name="opentelemetry", telemetry.sdk.version="1.45.0"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=org.apache.pulsar.broker, version=null, schemaUrl=null, attributes={}}, name=pulsar.broker.producer.message.incoming.count, description=The total number of messages received from this producer., unit={message}, type=LONG_SUM, data=ImmutableSumData{points=
...
at org.apache.pulsar.broker.stats.BrokerOpenTelemetryTestUtil.lambda$assertMetricLongSumValue$12(BrokerOpenTelemetryTestUtil.java:78)
at org.apache.pulsar.broker.stats.BrokerOpenTelemetryTestUtil.assertMetricLongSumValue(BrokerOpenTelemetryTestUtil.java:77)
at org.apache.pulsar.client.api.NonPersistentTopicTest.testProducerRateLimit(NonPersistentTopicTest.java:424)
...(10 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)
at org.apache.pulsar.broker.stats.BrokerOpenTelemetryTestUtil.assertMetricLongSumValue(BrokerOpenTelemetryTestUtil.java:77)
at org.apache.pulsar.client.api.NonPersistentTopicTest.testProducerRateLimit(NonPersistentTopicTest.java:424)
at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
at java.base/java.lang.reflect.Method.invoke(Method.java:580)
at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
at org.testng.internal.invokers.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:47)
at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:76)
at org.testng.internal.invokers.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
at java.base/java.lang.Thread.run(Thread.java:1583)
```

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

--- a/pulsar-broker/src/test/java/org/apache/pulsar/client/api/NonPersistentTopicTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/client/api/NonPersistentTopicTest.java
@@ -371,9 +371,8 @@ public class NonPersistentTopicTest extends ProducerConsumerBase {
             stopBroker();
             startBroker();
             // produce message concurrently
-            final int totalProduceMessages = 10;
             @Cleanup("shutdownNow")
-            ExecutorService executor = Executors.newFixedThreadPool(totalProduceMessages);
+            ExecutorService executor = Executors.newFixedThreadPool(10);
             AtomicBoolean failed = new AtomicBoolean(false);
             @Cleanup
             Consumer<byte[]> consumer = pulsarClient.newConsumer().topic(topic).subscriptionName("subscriber-1")
@@ -382,12 +381,11 @@ public class NonPersistentTopicTest extends ProducerConsumerBase {
             @Cleanup
             Producer<byte[]> producer = pulsarClient.newProducer().topic(topic).producerName(producerName).create();
             byte[] msgData = "testData".getBytes();
+            final int totalProduceMessages = 500;
             CountDownLatch latch = new CountDownLatch(totalProduceMessages);
-            final CyclicBarrier barrier = new CyclicBarrier(totalProduceMessages);
             for (int i = 0; i < totalProduceMessages; i++) {
                 executor.submit(() -> {
                     try {
-                        barrier.await();
                         producer.send(msgData);
                     } catch (Exception e) {
                         log.error("Failed to send message", e);

